### DISCUSSION TITLE
Physx collision trigger Events are lost.

### DISCUSSION THREAD

**Discussion by soltsmank** [o3de/o3de#6941] [Help Wanted]
Hello O3DEneers!

While testing a small game logic in ScriptCanvas, I paid attention that events are probably lost somehow.

Issue:
There are 2 spawned prefabs in a level. One physx collider with trigger enabled and can collide with both of spawned prefabs.
Both prefabs after some time collide and trigger is fired.
Looks like the trigger works only for single collision at a moment, so the script that listens to the trigger event only gets executed once, should be 2.
(To workaround it, I add several small physx colliders instead of one big area collider)
On trigger collision event I am deactivating collided prefab and firing event (INIT_SPAWN) to initiate spawn of a new prefab.
When INIT_SPAWN received, it calls for Delay node (for 3 seconds) , then 'onDone' sends SPAWNING event.
As a result,  I am left with 1 spawned new prefab, instead of 2.

I paid attention it happens only when the trigger collision happens for both prefabs at the same time.

Probably, one of the nodes or ScriptEvent doesn't support queue of events.

**Any clues where could be a problem?**
(No Lua scripts, just scriptcanvas scripting and scriptevent as a message queue bus)

**Comment by hultonha** [comment]
Hi @soltsmank, it sounds possibly like this might be more of an issue than a general discussion so it might be worth creating a separate issue to track with precise reproduction steps (and a test level/scene if you don't mind sharing it and zipping it up). We are pretty good at triaging incoming issues and responding (though not always at getting a precise fix as we can't commit to a time frame I'm afraid). Let us know if you have any other questions of if there's anything we can help with 👍
